### PR TITLE
support for non-contiguous index selection along axis

### PR DESCRIPTION
**Pull request by pegasos1**
Wrote some tests for the implementation we discussed. I thought about handling errors around `stack`, but index errors will crash the program in `isubview` beforehand anyway.

### CODE REVIEW

File: src/impl_methods.rs [hunk 1]
```diff
@@ -196,6 +197,26 @@ impl<A, S, D> ArrayBase<S, D> where S: Data<Elem=A>, D: Dimension
         debug_assert!(self.pointer_is_inbounds());
     }
 
+    pub fn select_along(&self, axis: Axis, indices: &[Ix]) -> OwnedArray<A, D>
```
Comment (bluss) [inline src/impl_methods.rs]: I think select is a good keyword (probably got it from daniel's issue). Do you have a suggestion for an even better method name here? I was thinking that this is maybe only one kind of such a method, this is the along an axis selection. `select_along` would suggest "along an axis" maybe `select_axis` is better? Only `select`? Something else.

This method is actually in the wrong place. It's not so important that it should be before .get() in the docs. I guess it's not a perfectly sorted order. Below the subview methods seems ok.

**Comment by pegasos1** [comment]
I think `select` makes sense. This type of indexing is generic enough. Also, any indexing along 2 or more axes is probably more suited to methods like get().

whoops, forgot to move that! Also, i'll add some basic docs.

**Comment by bluss** [comment]
Btw, the PR says it has a merge conflict. Select sounds good. @daniel-vainsencher is very welcome with feedback also after the PR is merged, should be no trouble to change things around to better names.

docs are great. Notice that one can select the same view multiple times and that's just fine I think.

**Comment by pegasos1** [comment]
oh, we could do `select_subview` to keep consistent with other `subview` methods. 
I'll look into those conflicts.

**Comment by bluss** [comment]
hm that sounds pretty good. with an s in subviews?

**Comment by pegasos1** [comment]
nah I was thinking singular subview. why make it plural?

**Comment by bluss** [comment]
Since it picks several subviews and concats them it seems appropriate. Even if the terminology is ad hoc.

**Comment by bluss** [comment]
It's a bit long though. Naming things..

**Comment by pegasos1** [comment]
haha yeah, probably the hardest part of programming.

**Comment by pegasos1** [comment]
if you want, we can leave it at `select`, and then change it if we want later.

**Comment by bluss** [comment]
seems best for now

**Comment by pegasos1** [comment]
good to go

**Comment by bluss** [comment]
thanks! this is a good feature. I'll edit the docs slightly.

**Pull request merged**